gammapy/data/event_list.py

```diff
@@ -481,19 +481,20 @@ def check(self, checks="all"):
         checker = EventListChecker(self)
         return checker.run(checks=checks)
 
-    def _get_coord_from_geom(self, geom):
-        """Return MapCoord of the EventList relative to an input geometry.
+    def map_coord(self, geom):
```

Comment:
I think the `EventList` API would be more consistent, if we don't pass `geom` and make `map_coord` a property just as `.time`, `.offset`, `.skycoord`. Just bundle all relevant coordinates and avoid copies if we are concerned about efficiency...

Reply:
Did you see this?

https://github.com/gammapy/gammapy/pull/2468/files#diff-0e905bc71b734b5e047ddf9dce8d2ef6R59-R64

@registerrier already implemented logic to match events table to arbitrary map axis, e.g. for `EVENT_ID`, if the names match in a case insensitive way.

Removing the `geom` argument here would remove that feature, or reduce it to a hard-coded preset list of coords.

So I'll leave as-is for now, I still think this PR is a step in the right direction.
But yes, possibly this is something that could be changed in the future, probably we should only decide once we have real-world use cases (i.e. a data reduction and fittting working for maps with a TIME or  EVENT_ID axis).

IF the conclusion is that `geom` shouldn't be passed to `events.map_coord`, then I agree that `map.fill_events` isn't useful any more - it'll be easy to replace, because you can just right-click in PyCharm and say "list callers" or use grep.